include/arch/riscv/arch/32/mode/machine.h

```diff
@@ -6,6 +6,42 @@
 
 #pragma once
 
-/* Place holder for 32-bit machine header */
-
+#include <util.h>
 #include <arch/model/smp.h>
+#include <stdint.h>
+
+static inline uint64_t riscv_read_time(void)
+{
+    register word_t nH_prev, nH, nL;
+    asm volatile(
+        "rdtimeh %0\n"
+        "rdtime  %1\n"
+        "rdtimeh %2\n"
+        : "=r"(nH_prev), "=r"(nL), "=r"(nH)
+        : /* no input */
+        : "memory");
+    if (nH_prev != nH) {
+        /* Ensure that the time is correct if there is a rollover in the
+         * high bits between reading the low and high bits. */
+        asm volatile("rdtime  %0\n" : "=r"(nL) :: "memory");
+    }
+    return ((uint64_t)nH << 32) | nL;
+}
+
+static inline uint64_t riscv_read_cycle(void)
+{
+    register word_t nH_prev, nH, nL;
+    asm volatile(
+        "rdcycleh %0\n"
+        "rdcycle  %1\n"
+        "rdcycleh %2\n"
+        : "=r"(nH_prev), "=r"(nL), "=r"(nH)
+        : /* no input */
+        : "memory");
+    if (nH_prev != nH) {
+        /* Ensure that the cycles are correct if there is a rollover in the
+         * high bits between reading the low and high bits. */
+        asm volatile("rdcycle  %0\n" : "=r"(nL) :: "memory");
```

Comment:
this isn't a clean move - you've changed the code - can you please seperate the move into a different commit? it's nearly impossible to review if code is moved and changed in one commit